src/shared/extensions/yaml.ts

```diff
@@ -76,8 +115,10 @@ export async function activateYamlExtension(): Promise<YamlExtension | undefined
         }
     )
     return {
-        assignSchema: (path, schema) => schemaMap.set(path.toString(), applyScheme(AWS_SCHEME, evaluate(schema))),
-        removeSchema: path => schemaMap.delete(path.toString()),
-        getSchema: path => schemaMap.get(path.toString()),
+        assignSchema: (path, registry, schema) => {
+            registerSchema(path, registry, schema, schemaMap)
```

Comment:
"assign" and "register" are approximately synonyms. This is a signal that the API needs more work. However, to unblock this PR, could for now add one-line comments explaining the difference.